packages/flutter_tools/lib/src/commands/build_web.dart

```diff
@@ -87,6 +96,19 @@ class BuildWebCommand extends BuildSubCommand {
     if (buildInfo.isDebug) {
       throwToolExit('debug builds cannot be built directly for the web. Try using "flutter run"');
     }
+    if (stringArg('base-href') != null && !(stringArg('base-href').startsWith('/') && stringArg('base-href').endsWith('/'))) {
+      throwToolExit('base-href should start and end with /');
+    }
+    if (!globals.fs.currentDirectory
+        .childDirectory('web')
+        .childFile('index.html')
+        .readAsStringSync()
+        .contains(r'$FLUTTER_BASE_HREF') &&
+        stringArg('base-href') != null) {
+      throwToolExit(
+          r'Placeholder $FLUTTER_BASE_HREF not found in base tag of index.html.'
+          r' Please set href attribute of base tag as $FLUTTER_BASE_HREF to use this flag.');
```

Comment:
``Couldn't find the placeholder for base href. ``
``Please add `<base href="$FLUTTER_BASE_HREF">` to web/index.html``